### PR TITLE
Download Plantuner from HTTPS URL and pin version by commit hash

### PR DESCRIPTION
**Pull request by gertvdijk**
Before this change:
- Any version on default HEAD would be taken and thus a non-reproducible
  build.
- Insecure download over old-style Git protocol.

I was unable to find a stable HTTPS-powered mirror of the upstream
repository git://www.sigaev.ru/plantuner.git, so I created my own on
GitHub.
I did consider https://github.com/postgrespro/plantuner, but found its
state debatable as expressed in an issue filed there:
https://github.com/postgrespro/plantuner/issues/1#issue-566917641.

Fixes zalando/spilo#384

### CODE REVIEW

File: postgres-appliance/Dockerfile [hunk 1]
```diff
@@ -96,7 +97,9 @@ RUN export DEBIAN_FRONTEND=noninteractive \
         && git clone -b $DECODERBUFS https://github.com/debezium/postgres-decoderbufs.git \
         && git clone -b $PLPROFILER https://github.com/bigsql/plprofiler.git \
         && git clone -b $PLPGSQL_CHECK https://github.com/okbob/plpgsql_check.git \
-        && git clone git://www.sigaev.ru/plantuner.git \
+        && mkdir plantuner \
+            && curl -sL https://github.com/gertvdijk/plantuner/archive/$PLANTUNER_COMMIT.tar.gz \
```
Comment (CyberDem0n) [inline postgres-appliance/Dockerfile]: I would better prefer keeping the tarball in the `postgres-appliance/dependencies/`

Comment (gertvdijk) [reply postgres-appliance/Dockerfile]: Okay, I will change that. :) Thank for your review.

**Comment by avaczi** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**